ql/src/java/org/apache/hadoop/hive/ql/parse/repl/load/FailoverMetaData.java

```diff
@@ -0,0 +1,197 @@
+        this.cursorPoint = otherDMD.cursorPoint;
+        this.txnsWithoutLocks = otherDMD.txnsWithoutLocks;
+        this.initialized = true;
+    }
+
+    private void loadMetadataFromFile() throws SemanticException {
+        LOG.info("Reading failover metadata from file: ", dumpFile);
+        BufferedReader br = null;
+        try {
+            FileSystem fs = dumpFile.getFileSystem(hiveConf);
+            br = new BufferedReader(new InputStreamReader(fs.open(dumpFile)));
+            String line;
+            if ((line = br.readLine()) != null) {
+                FailoverMetaData otherDMD = JSON_OBJECT_MAPPER.readValue(line, FailoverMetaData.class);
+                setMetaData(otherDMD);
+            } else {
+                throw new IOException("Unable to read valid values from failover Metadata file:"
+                        + dumpFile.toUri().toString());
+            }
+
+        } catch (IOException ioe) {
+            throw new SemanticException(ioe);
+        } finally {
+            if (br != null) {
+                try {
+                    br.close();
+                } catch (IOException e) {
+                    throw new SemanticException(e);
+                }
+            }
+        }
+    }
+
+    public void setEventId(Long eventId) {
+        this.eventId = eventId;
+        initialized = true;
+    }
+
+    public void setAbortedTxns(List<Long> abortedTxns) {
+        this.abortedTxns = abortedTxns;
+        initialized = true;
+    }
+
+    public void addToAbortedTxns(List<Long> abortedTxns) {
+        if (this.abortedTxns == null) {
+            this.abortedTxns = abortedTxns;
+        } else {
+            this.abortedTxns.addAll(abortedTxns);
+        }
+    }
+
+    public void setCursorPoint(Long cursorPoint) {
+        this.cursorPoint = cursorPoint;
+        initialized = true;
+    }
+
+    public void setOpenTxns(List<Long> openTxns) {
+        this.openTxns = openTxns;
+        initialized = true;
+    }
+
+    public void setTxnsWithoutLocks(List<Long> txnsWithoutLocks) {
+        this.txnsWithoutLocks = txnsWithoutLocks;
+        initialized = true;
+    }
+
+    public Long getCursorPoint() throws SemanticException {
+        initializeIfNot();
+        return cursorPoint;
+    }
+
+    public List<Long> getOpenTxns() throws SemanticException {
+        initializeIfNot();
+        return openTxns;
+    }
+
+    public Long getEventId() throws SemanticException {
+        initializeIfNot();
+        return eventId;
+    }
+
+    public List<Long> getAbortedTxns() throws SemanticException {
+        initializeIfNot();
+        return abortedTxns;
+    }
+
+    public List<Long> getTxnsWithoutLocks() throws SemanticException {
+        initializeIfNot();
+        return txnsWithoutLocks;
+    }
+
+    @JsonIgnore
+    public boolean isValidMetadata() throws SemanticException {
+        initializeIfNot();
+        return openTxns != null && abortedTxns != null && eventId != null
```

Comment:
What would be values for openTxns/abortedTxns if there is no txn which were open and hence nothing was aborted?

Reply:
empty list(Not null)